src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,51 @@
+body {
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: $font-size-day;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-day;
+  padding: $paddings-calendar;
+  width: getCalendarSize(
+    $length-day,
+    $gap-day,
+    $colums-calendar,
+    $paddings-calendar
+  );
+
+  @each $day, $start in $week-days {
+    &--start-day-#{$day} :first-child {
+      margin-left: $start;
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} :nth-child(n + #{$i + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    @include square-size-day;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: $color-day;
+    border: 1px solid black;
+    transition: transform 0.5s, background-color 0.5s;
+  }
+
+  @for $i from 1 through 31 {
+    &__day:nth-child(#{$i})::before {
+      content: "#{$i}";
+    }
+  }
+
+  &__day:hover {
+    cursor: pointer;
+    background-color: pink;
+    transform: translateY(-20px);
```

Comment:
[STYLES] - Don't be greedy, add some top paddings, so the top row don't go over top of the page on hover.
The boxes of the first row go outside of the body.